crates/oxc_minifier/src/printer/gen.rs

```diff
@@ -502,7 +502,13 @@ impl<'a> Gen for FunctionBody<'a> {
         for directive in &self.directives {
             directive.gen(p);
         }
-        p.needs_semicolon = false;
+        p.needs_semicolon =
+            if let Some(Statement::ExpressionStatement(expr_stmt)) = self.statements.get(0)
+            && let Expression::StringLiteral(_) = &expr_stmt.expression {
+                true
+            } else {
+                false
+            };
         for stmt in &self.statements {
```

Comment:
you get add an index here, self.statements.iter().enumerate(), then check i == 0 and stmt == string

Reply:
I wonder if only checking once in the outer of for loop is a more effective way.

Reply:
> I wonder if only checking once in the outer of for loop is a more effective way.

Good point, both are fine by me.